internal/storetesting/entities.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2014 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storetesting
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/charm.v5-unstable"
+	"gopkg.in/mgo.v2"
+
+	"gopkg.in/juju/charmstore.v4/internal/mongodoc"
+)
+
+// EntityBuilder provides a convenient way to describe a mongodoc.Entity
+// for tests that is correctly formed and contain the desired
+// information.
+type EntityBuilder struct {
+	entity *mongodoc.Entity
+}
+
+// NewEntity creates a new EntityBuilder for the provided URL.
+func NewEntity(url string) EntityBuilder {
+	URL := charm.MustParseReference(url)
+	return EntityBuilder{
+		entity: &mongodoc.Entity{
+			URL:                 URL,
+			Name:                URL.Name,
+			Series:              URL.Series,
+			Revision:            URL.Revision,
+			User:                URL.User,
+			BaseURL:             baseURL(URL),
+			PromulgatedRevision: -1,
+		},
+	}
+}
+
+// PromulgatedURL sets the PromulgatedURL and PromulgatedRevision of the
+// entity being built.
+func (b EntityBuilder) PromulgatedURL(url string) EntityBuilder {
```

Comment:
This seems more a getter than a setter. Maybe just Promulgate(url string)? [shrug]